### PR TITLE
Proxy request.signal through in vite dev

### PR DESCRIPTION
**Pull request by brophdawg11**
Found this wasn't working right while testing https://github.com/remix-run/remix/pull/9975.  Aligned with how we do it in the express adapter.

Closes https://github.com/remix-run/remix/issues/9438

### CODE REVIEW

**Review by jacob-ebey — APPROVED** [review]

File: .changeset/hungry-experts-suffer.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@remix-run/dev": patch
+---
+
+Abort request.signal in vite dev when the node response is closed
```
Comment (brophdawg11) [inline .changeset/hungry-experts-suffer.md:5]:
```suggestion
Properly abort `request.signal` during `vite dev` when the node response is closed
```

**Comment by jvaill** [comment]
It's working better now! But I'm still having issues with the signal not being aborted sometimes.

Referencing this: https://github.com/sergiodxa/remix-utils/issues/317#issuecomment-2369224485

**Comment by Eiley2** [comment]
Is there a workaround for this? Maybe check every once is a while if a connection is closed to abort that connection? I'm having the same issue with node and the updated version of remix in production too.


sse.time.ts
```typescript
import { LoaderFunctionArgs } from "@remix-run/node"
import { eventStream } from "remix-utils/sse/server"

export async function loader({ request }: LoaderFunctionArgs) {
  return eventStream(request.signal, function setup(send) {
    const timer = setInterval(() => {
      console.log(new Date().toLocaleTimeString())
      send({ event: "time", data: new Date().toLocaleTimeString() })
    }, 1000)

    return function clear() {
      console.log("SSE connection closed")
      clearInterval(timer)
    }
  })
}
``` 

and this is my route:


```typescript

export default Route(){
  let time = useEventSource("/sse/time", { event: "time" })
  if (!time) {
    time = new Date().toLocaleTimeString()
  }

return <p>{time}</p>
```

**Comment by jvaill** [comment]
I tracked this down to some weirdness in undici.

It seems like the signal is working correctly until a new undici `Request()` is constructed. At that point, undici creates an underlying signal. That ends up getting passed to the remix handler.

[Here's where `Request()` is created in remix.](https://github.com/remix-run/remix/blob/main/packages/remix-server-runtime/data.ts#L86)

[Here's an issue I opened on the undici repo with a minimal test case.](https://github.com/nodejs/undici/issues/3644)

**Comment by jvaill** [comment]
This should fix the remaining issues: https://github.com/remix-run/remix/pull/10030

cc @Eiley2

**Comment by jvaill** [comment]
Possible fix: https://github.com/remix-run/remix/pull/10030

**Comment by mkromann** [comment]
I am running into the same issue, and cannot get any EventSource to close with request.signal. Is there a workaround for how to handle this?

Right now it is a huge block for developing with SSE. We have to restart the whole app constantly.

**Pull request merged**